esmf_regrid/tests/unit/experimental/unstuctured_regrid/test_MeshInfo.py

```diff
@@ -85,3 +85,75 @@ def _give_extra_dims(array):
     extra_dim_mesh_output = grid_to_mesh_regridder.regrid(extra_dim_grid_input)
     extra_dim_expected_mesh_output = _give_extra_dims(expected_mesh_output)
     assert ma.allclose(extra_dim_expected_mesh_output, extra_dim_mesh_output)
+
+
+def test_regrid_bilinear_with_mesh():
+    """Basic test for regridding with :meth:`~esmf_regrid.esmf_regridder.GridInfo.make_esmf_field`."""
+    # We create a mesh with the following shape:
+    # 20     ,+                  ,+ 4
+    #       / |  with nodes:    / |
+    # 10  +' ,+             1 +' ,+ 3
+    #     | / |               | / |
+    #  0  +---+             0 +---+ 2
+    #     0  10
+    mesh_args = _make_small_mesh_args()
+    elem_coords = np.array([[5, 0], [5, 10]])
+    node_mesh = MeshInfo(*mesh_args, location="node")
+    face_mesh = MeshInfo(*mesh_args, elem_coords=elem_coords, location="face")
+
+    # We create a grid with the following shape:
+    # 20/3 +---+
+    #      |   |
+    # 10/3 +---+
+    #      |   |
+    #  0   +---+
+    #      0  10
+    grid_args = [ar * 2 for ar in make_grid_args(2, 3)]
+    grid = GridInfo(*grid_args, center=True)
+
+    mesh_to_grid_regridder = Regridder(node_mesh, grid, method="bilinear")
+    mesh_input = np.arange(5)
+    grid_output = mesh_to_grid_regridder.regrid(mesh_input)
+    # For a flat surface, we would expect the fractional part of these values
+    # to be either 1/3 or 2/3. Since the actual surface lies on a sphere, and
+    # due to the way ESMF approximates these values, the expected output is
```

Comment:
Perhaps "calculates" is a better word than "approximates". I guess my point is that the values will be close to our intuition for flat surfaces but will necessarily be different since we are not flat. You could view this as a way of approximating the behaviour of flat bilinear regridding, but I don't think I can say that ESMF is _wrong_ here, since I'm not entirely sure what the correct behaviour _ought_ to be in this case. Since the numbers are close enough to my intuition, I'm giving ESMF the benefit of the doubt that the "approximation" here is an appropriate one.